### ISSUE TITLE
Only audio track is playing / black screen and no video

### ISSUE THREAD

**Issue by matthewhilton** [catalyst/moodle-filter_smartmedia#33]
**TLDR**
We discovered that a site suddenly started only playing the audio for smartmedia videos, and not the video. As it turns out, this is because of the 'bandwidth' as perceived by the browser when accessing the site.

Essentially the videojs library was determining that the browser did not have enough bandwidth to play the video track, and only played the audio track.

Related: https://github.com/videojs/http-streaming/issues/846

## Background

- The video files output by AWS elastic transcoder used by smartmedia aren't normal video files, they are video _playlists_ - these are .m3u8 and .mdp (note mdp is not supported by firefox natively)
- Video playlists contain a number of audio and video 'tracks' with different bitrates. The idea is that the video player selects an appropriate track based on the users connection.
- Its important to note that video playlists afaik are NOT supported natively by browsers, and so videojs is used and has a built in library to play these files.

## Details
This issue appeared initially on a production site, but when I tried to recreate it locally it always appeared to work fine.

I checked the network tab while playing the video, and noticed that on the prod site only 1 .ts file was being downloaded, whereas on my local 2 .ts files were being loaded.

For context, .ts files are used to stream audio/video. Essentially the .m3u8 file contains a bunch of tracks, and when you load these tracks it streams a bunch of .ts files (this is all handled by the videojs library)

And so I hypothesised that the prod site was only streaming 1 .ts file (the audio) and local was streaming 2 .ts files (the audio and video). 

This **was not** caused by a bad conversion - the prod site HAD both tracks (for e.g. if you use https://addons.mozilla.org/en-US/firefox/addon/native-mpeg-dash-hls-playback/ the video plays fine with audio and video)

Digging into the video-js library in Moodle (video-lazy.js) there is a `systemBandwidth` function that calculates the bandwidth used to select which track to play: https://github.com/moodle/moodle/blob/6307f8dd89bcf3c057b2bfd0518a207131b4bf5a/media/player/videojs/amd/src/video-lazy.js#L57689

I found I could recreate the issue locally by editing this and just returning a low number, e.g. 100 (normally > 100k bits per second). This recreated the issue perfectly on my local machine - only the audio would play and the video would never play.

Now that I had recreated the issue, I tried turning on `enableLowInitialPlaylist` as described in https://github.com/videojs/http-streaming/issues/846 - this worked and got the video to always play even with my systemBandwidth faked to be very low.

## Fixes
Theres a number of ways this could be fixed:

1. edit `systemBandwidth` to always return some high number -> this is pointless because it negates all the benefits of even using playlists with multiple bitrates
2. Turn on low and medium quality in `local_smartmedia` -> this isn't ideal since these might be turned off because the quality is not very good. Turning these on _does_ kind of solve the issue because their bitrates are lower, and so are more likely to get selected. But ideally we always want a video track to be selected, even if the connection is horrible and only high quality is available.
3. Pass in `enableLowInitialPlaylist` - this is the most ideal option.  Enabling this changes the playlist selector function from `lastBandwidthSelector` to `lowestBitrateCompatibleVariantSelector` - `lowestBitrateCompatibleVariantSelector` chooses the lowest bitrate playlist with video https://github.com/moodle/moodle/blob/6307f8dd89bcf3c057b2bfd0518a207131b4bf5a/media/player/videojs/amd/src/video-lazy.js#L51151-L51154 - after which the video starts playing it will then switch the playlist to a higher bitrate (if the connection is good)

I will be making a PR for fix 3 very shortly.
